controllers/operands/kubevirt.go

```diff
@@ -291,6 +293,20 @@ func hcWorkloadUpdateStrategyToKv(hcObject *hcov1beta1.HyperConvergedWorkloadUpd
 	return kvObject
 }
 
+func filterWorkloadupdatemethods(workloadUpdateMethods []string) []string {
+	var filterWorkloadUpdateMethods []string
+	for i := range workloadUpdateMethods {
+		if workloadUpdateMethods[i] == string(kubevirtcorev1.WorkloadUpdateMethodLiveMigrate) {
+			if hcoutil.GetClusterInfo().IsInfrastructureHighlyAvailable() {
+				filterWorkloadUpdateMethods = append(filterWorkloadUpdateMethods, workloadUpdateMethods[i])
+			}
+		} else {
+			filterWorkloadUpdateMethods = append(filterWorkloadUpdateMethods, workloadUpdateMethods[i])
+		}
+	}
+	return filterWorkloadUpdateMethods
+}
```

Comment:
What do you think about this?

```suggestion
        if hcoutil.GetClusterInfo().IsInfrastructureHighlyAvailable() {
            copy(workloadUpdateMethods, workloadUpdateMethods)
        } else {
            for _, method := range workloadUpdateMethods {
                if method != string(kubevirtcorev1.WorkloadUpdateMethodLiveMigrate) {
                    filterWorkloadUpdateMethods = append(filterWorkloadUpdateMethods, method)
                }
            }
        }
	return filterWorkloadUpdateMethods
}
```

Reply:
nice, and we can even do better avoiding the copy at all if not on SNO